uberfire-m2repo-editor/uberfire-m2repo-editor-backend/src/main/java/org/guvnor/m2repo/backend/server/M2ServletContextListener.java

```diff
@@ -0,0 +1,195 @@
+    }
+
+    @Override
+    public void contextDestroyed(ServletContextEvent servletContextEvent) {
+    }
+
+    private int deployJarsFromWar(String path) {
+        int i = 0;
+        try (DirectoryStream<Path> ds = Files.newDirectoryStream(Paths.get(path))) {
+            RepositorySystemSession session = newSession(newRepositorySystem());
+            for (Path p : ds) {
+                if (p.toString().endsWith(JAR_EXT)) {
+                    deployJar(p.toAbsolutePath().toString(),
+                              session);
+                    i++;
+                }
+            }
+        } catch (Exception e) {
+            logger.error(e.getMessage(),
+                         e);
+        }
+        return i;
+    }
+
+    public GAV deployJar(String file,
+                         RepositorySystemSession session) {
+        GAV gav = new GAV();
+        Properties props = readZipFile(file);
+        if (!props.isEmpty()) {
+            gav = new GAV(props.getProperty(GROUP_ID),
+                          props.getProperty(ARTIFACT_ID),
+                          props.getProperty(VERSION));
+            deploy(gav,
+                   file,
+                   session);
+        }
+        return gav;
+    }
+
+    public Properties readZipFile(String zipFilePath) {
+        try {
+            ZipFile zipFile = new ZipFile(zipFilePath);
+            Enumeration<? extends ZipEntry> e = zipFile.entries();
+            while (e.hasMoreElements()) {
+                ZipEntry entry = e.nextElement();
+                String end = "pom.properties";
+                if (!entry.isDirectory() && entry.getName().startsWith(MAVEN_META_INF) && entry.getName().endsWith(end)) {
+                    BufferedInputStream bis = new BufferedInputStream(zipFile.getInputStream(entry));
+                    Properties props = new Properties();
+                    props.load(bis);
+                    bis.close();
+                    return props;
+                } else {
+                    continue;
+                }
+            }
+        } catch (IOException e) {
+            logger.error("IOError :{}",
+                         e.getMessage(),
+                         e);
+        }
+        return new Properties();
+    }
+
+    public boolean deploy(GAV gav,
+                          String jarFile,
+                          RepositorySystemSession session) {
+        Artifact jarArtifact = new DefaultArtifact(gav.getGroupId(),
+                                                   gav.getArtifactId(),
+                                                   JAR_ARTIFACT,
+                                                   gav.getVersion());
+        jarArtifact = jarArtifact.setFile(new File(jarFile));
+        try {
+            final InstallRequest installRequest = new InstallRequest();
+            installRequest.addArtifact(jarArtifact);
+            InstallResult result = Aether.getAether().getSystem().install(session,
```

Comment:
not sure if the getSystem here will be resolved to ArtifactRepositoryService.GLOBAL_M2_REPO_NAME